### PR TITLE
[fix](like_func) incorrect result of like with 'NO_BACKSLASH_ESCAPES' mode

### PR DESCRIPTION
**Pull request by mrhhsg**
## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by mrhhsg** [comment]
run buildall

**Comment by mrhhsg** [comment]
run buildall

File: be/src/vec/functions/like.cpp [hunk 1]
```diff
@@ -317,6 +317,8 @@ Status FunctionLikeBase::hs_prepare(FunctionContext* context, const char* expres
         return Status::RuntimeError("hs_alloc_scratch allocate scratch space error");
     }
 
+    LOG(INFO) << "****** expression: " << expression;
```
Comment (yiguolei) [inline be/src/vec/functions/like.cpp:320]: maybe  too many logs

**Review by yiguolei — APPROVED** [review]
LGTM

**Comment by yiguolei** [comment]
run buildall

**Review by yiguolei — DISMISSED** [review]
LGTM

**Comment by mrhhsg** [comment]
run buildall

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by yiguolei — APPROVED** [review]
LGTM

File: be/src/vec/functions/like.cpp [hunk 2]
```diff
@@ -453,24 +453,30 @@ void FunctionLike::convert_like_pattern(LikeSearchState* state, const std::strin
...
+            default:
+                is_escaped = pattern[i] == state->escape_char;
+                if (!is_escaped) {
+                    re_pattern->append(1, pattern[i]);
+                }
+                break;
+            }
         } else {
-            // regular character or escaped special character
+            if (pattern[i] == '.' || pattern[i] == '[' || pattern[i] == ']' || pattern[i] == '{' ||
```
Comment (zhiqiang-hhhh) [inline be/src/vec/functions/like.cpp:472]: these characters have special meaning in re2 so they must be escaped

**Pull request merged**